### PR TITLE
nimdow: init at 0.7.36

### LINKED ISSUE #178866: Nimdow Window Manager Package Request
**Project description**
The Nimdow Window Manager is a wm built in the Nim programming language. It is similar to DWM and i3. 

**Metadata**

* homepage URL: https://github.com/avahe-kellenberger/nimdow
* source URL: https://github.com/avahe-kellenberger/nimdow
* license: GPL-2.0
* platforms: Linux (and possibly others?)

### PR DESCRIPTION
**Pull request by marcusramberg**
###### Description of changes

Package nimdow 0.7.36 as well as required dependencies in nimPackages Fixes #178866

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by marcusramberg** [comment]
ps: Have just assumed the license for safeset, will try to get it figured out soon.

**Comment by marcusramberg** [comment]
> ps: Have just assumed the license for safeset, will try to get it figured out soon.

image

File: pkgs/top-level/nim-packages.nix [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 lib.makeScope newScope (self:
   let callPackage = self.callPackage;
-  in {
+  in
+  {
```
Comment (stigtsp) [inline pkgs/top-level/nim-packages.nix:6]: Some unintentional reformatting maybe

File: pkgs/top-level/nim-packages.nix [hunk 2]
```diff
@@ -37,7 +38,8 @@ lib.makeScope newScope (self:
     frosty = callPackage ../development/nim-packages/frosty { };
 
     getdns = callPackage ../development/nim-packages/getdns {
-      inherit (pkgs) getdns; };
+      inherit (pkgs) getdns;
+    };
```
Comment (stigtsp) [inline pkgs/top-level/nim-packages.nix:42]: Some unintentional reformatting maybe

File: nixos/modules/services/x11/window-managers/nimdow.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.xserver.windowManager.nimodw;
+in
+{
+  ###### interface
```
Comment (stigtsp) [inline nixos/modules/services/x11/window-managers/nimdow.nix:9]:
```suggestion
```

File: nixos/modules/services/x11/window-managers/nimdow.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  options = {
+    services.xserver.windowManager.nimdow.enable = mkEnableOption (lib.mdDoc "nimdow");
+  };
+
+  ###### implementation
```
Comment (stigtsp) [inline nixos/modules/services/x11/window-managers/nimdow.nix:14]:
```suggestion
```

**Review by stigtsp — CHANGES_REQUESTED** [review]
Hi! Some nitpick:
- Pls avoid (unintentional) reformatting in nim-packages.nix
- Comments are probably not useful in services/x11/window-managers/nimdow.nix module

File: pkgs/top-level/nim-packages.nix (hunk 2 above)
Comment (marcusramberg) [reply pkgs/top-level/nim-packages.nix]: pushed fix.

File: pkgs/top-level/nim-packages.nix (hunk 1 above)
Comment (marcusramberg) [reply pkgs/top-level/nim-packages.nix]: pushed fix.

File: nixos/modules/services/x11/window-managers/nimdow.nix (hunk 4 above)
Comment (marcusramberg) [reply nixos/modules/services/x11/window-managers/nimdow.nix]: pushed fix.

**Review by stigtsp — DISMISSED** [review]

**Review by ghost — APPROVED** [review]

File: nixos/modules/services/x11/window-managers/nimdow.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{ config, lib, pkgs, ... }:
```
Comment (ehmry) [inline nixos/modules/services/x11/window-managers/nimdow.nix:1]: This module should be added to `nixos/modules/module-list.nix`.

Comment (ehmry) [reply nixos/modules/services/x11/window-managers/nimdow.nix:1]: It should also be documented as a new NixOS feature in the release notes (`./nixos/doc/manual/release-notes/rl-2305.section.md`).

**Review by ghost — CHANGES_REQUESTED** [review]

File: nixos/modules/services/x11/window-managers/nimdow.nix (hunk 5 above)
Comment (marcusramberg) [reply nixos/modules/services/x11/window-managers/nimdow.nix:1]: Thanks. I've updated the module commit with these two changes now.

**Review by stigtsp — APPROVED** [review]
LGTM

**Review by ghost — APPROVED** [review]

**Pull request merged**